src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java

```diff
@@ -374,6 +374,17 @@ public static String getPathWithoutBucket(String path) {
 
     }
 
+    /**
+     * String -> Path. This *should* not be necessary (use Paths.get(URI.create(...)) instead) , but it currently is
+     * on Spark because using the fat, shaded jar breaks the registration of the GCS FilesystemProvider.
```

Comment:
There doesn't seem to be an easy way around this. It looks related to the way Spark (YARN) starts processes. I could find an [identical question](http://stackoverflow.com/questions/38728479/how-can-i-use-a-jsr-203-file-system-provider-with-apache-spark) on StackOverflow but it has no answer.